src/dist/mod.rs

```diff
@@ -208,6 +208,17 @@ impl fmt::Display for PartialVersion {
 impl FromStr for PartialVersion {
     type Err = anyhow::Error;
     fn from_str(ver: &str) -> Result<Self> {
+        // `semver::Comparator::from_str` supports an optional operator
+        // (e.g. `=`, `>`, `>=`, `<`, `<=`, `~`, `^`, `*`) before the
+        // partial version, so we should exclude that case first.
+        if let Some(ch) = ver.chars().nth(0) {
```

Comment:
Done!

Reply:
Honestly I prefer unit tests? (Sorry!)

* They're out of the way instead of in the middle of other code
* No having to deal with the peculiarities of doctests
* Better syntax highlighting (for one thing, in GitHub PRs)